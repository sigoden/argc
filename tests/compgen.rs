use super::SPEC_SCRIPT;

const HELP_TAG_SCRIPT: &str = r#"
# @help Print help information

# @cmd
foo() { :; }

# @cmd
bar() { :; }
"#;

#[test]
fn test_compgen() {
    snapshot_compgen!(SPEC_SCRIPT, "");
}

#[test]
fn test_compgen_help() {
    snapshot_compgen!(SPEC_SCRIPT, "help");
}

#[test]
fn test_compgen_subcommand() {
    snapshot_compgen!(SPEC_SCRIPT, "cmd_option_names ");
}

#[test]
fn test_compgen_option_choices() {
    snapshot_compgen!(SPEC_SCRIPT, "cmd_option_names --opt7 ");
}

#[test]
fn test_compgen_option_choices2() {
    snapshot_compgen!(SPEC_SCRIPT, "cmd_option_names --opt7 a");
}

#[test]
fn test_compgen_option_choices3() {
    snapshot_compgen!(SPEC_SCRIPT, "cmd_option_names --opt7 a ");
}

#[test]
fn test_compgen_positional() {
    snapshot_compgen!(SPEC_SCRIPT, "cmd_positional_requires ");
}

#[test]
fn test_compgen_positional_arg() {
    snapshot_compgen!(SPEC_SCRIPT, "cmd_positional_requires arg1 ");
}

#[test]
fn test_compgen_positional_arg2() {
    snapshot_compgen!(SPEC_SCRIPT, "cmd_positional_requires arg1 arg2 ");
}

#[test]
fn test_compgen_positional_choices() {
    snapshot_compgen!(SPEC_SCRIPT, "cmd_positional_with_choices ");
}

#[test]
fn test_compgen_help_tag() {
    snapshot_compgen!(HELP_TAG_SCRIPT, "");
}

#[test]
fn test_compgen_help_tag2() {
    snapshot_compgen!(HELP_TAG_SCRIPT, "help");
}

#[test]
fn test_compgen_choice_fn() {
    snapshot_compgen!(SPEC_SCRIPT, "cmd_option_names --op11 ");
}
