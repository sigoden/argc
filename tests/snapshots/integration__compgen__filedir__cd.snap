---
source: tests/compgen.rs
expression: data
---
************ COMPGEN `prog --oa ` ************
argc_value.rs	1	argc_value.rs		default
bin/	0	bin/		blue bold
build.rs	1	build.rs		default
command/	0	command/		blue bold
compgen.rs	1	compgen.rs		default
completions/	0	completions/		blue bold
lib.rs	1	lib.rs		default
mangen.rs	1	mangen.rs		default
matcher.rs	1	matcher.rs		default
param.rs	1	param.rs		default
parser.rs	1	parser.rs		default
runtime/	0	runtime/		blue bold
shell.rs	1	shell.rs		default
utils.rs	1	utils.rs		default

************ COMPGEN `prog --oa=` ************
--oa=argc_value.rs	1	argc_value.rs		default
--oa=bin/	0	bin/		blue bold
--oa=build.rs	1	build.rs		default
--oa=command/	0	command/		blue bold
--oa=compgen.rs	1	compgen.rs		default
--oa=completions/	0	completions/		blue bold
--oa=lib.rs	1	lib.rs		default
--oa=mangen.rs	1	mangen.rs		default
--oa=matcher.rs	1	matcher.rs		default
--oa=param.rs	1	param.rs		default
--oa=parser.rs	1	parser.rs		default
--oa=runtime/	0	runtime/		blue bold
--oa=shell.rs	1	shell.rs		default
--oa=utils.rs	1	utils.rs		default

************ COMPGEN `prog --ob ` ************
argc_value.rs:	0	argc_value.rs:		default
bin/	0	bin/		blue bold
build.rs:	0	build.rs:		default
command/	0	command/		blue bold
compgen.rs:	0	compgen.rs:		default
completions/	0	completions/		blue bold
lib.rs:	0	lib.rs:		default
mangen.rs:	0	mangen.rs:		default
matcher.rs:	0	matcher.rs:		default
param.rs:	0	param.rs:		default
parser.rs:	0	parser.rs:		default
runtime/	0	runtime/		blue bold
shell.rs:	0	shell.rs:		default
utils.rs:	0	utils.rs:		default

************ COMPGEN `prog foo=` ************
foo=argc_value.rs	1	argc_value.rs		default
foo=bin/	0	bin/		blue bold
foo=build.rs	1	build.rs		default
foo=command/	0	command/		blue bold
foo=compgen.rs	1	compgen.rs		default
foo=completions/	0	completions/		blue bold
foo=lib.rs	1	lib.rs		default
foo=mangen.rs	1	mangen.rs		default
foo=matcher.rs	1	matcher.rs		default
foo=param.rs	1	param.rs		default
foo=parser.rs	1	parser.rs		default
foo=runtime/	0	runtime/		blue bold
foo=shell.rs	1	shell.rs		default
foo=utils.rs	1	utils.rs		default
